design-patterns/src/main/java/decorator/PersistentSender.java

```diff
@@ -0,0 +1,26 @@
+package decorator;
+
+public class PersistentSender extends SenderDecorator {
+    private static int attempts = 0;
+
+    public PersistentSender(Sender sender) {
```

Comment:
please, don't put any logic inside the constructor, it is a bad idea in general

in this case, enriched logic should be around delegated method, not in the constructor